Romina.Api/Controllers/ProductController.cs

```diff
@@ -1,19 +1,56 @@
-﻿using Microsoft.AspNetCore.Mvc;
-using System;
+﻿using System;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc;
+using Romina.Api.Models;
+using Romina.Api.Repositories;
 
 namespace Romina.Api.Controllers
 {
+    [Route("api/product")]
     public class ProductController : Controller
     {
+        private readonly IProductRepository _productRepository;
+
+        public ProductController(IProductRepository productRepository)
+        {
+            _productRepository = productRepository;
+        }
+
         [HttpGet("{id}")]
-        public ActionResult<string> Get(string id)
+        public ActionResult<Product> Get(string id)
+        {
+            var product = _productRepository.GetProductById(id);
+            if (product == null)
+            {
+                return new NotFoundResult();
+            }
+
+            return product;
+        }
+
+        [HttpGet("{make}")]
+        //should this method always be of type actionresult? Or can I have it return List<Product>
+        public ActionResult<Product> GetByMake(string make)
```

Comment:
Hey bud so there's couple of things we need to change. 

I specified the endpoint in the acceptance criteria (on the readme). The endpoint I wanted you to create was the following `api/product?filter=nike shoes` ... so the new endpoints `GetByMake` or `GetByModel` aren't quite what we want.  

What you also need to consider is that we have unfortunately created a bug. When you run this API and hit the following `api/product/somevalue` you will get a `AmbiguousMatchException` (run the API and test this for yourself). The reason why this occurs is because all three endpoints on this product controller are effectively set up the same, they are **all** of type `GET` and take in a `string` as a parameter. 

So when you call the endpoint with `api/product/somevalue', the API doesn't know where to route the request. To give you an example, if you searched 'api/product/nike' how would we know if `nike` was a id, make or model? I hope this makes sense but if it doesn't lemme know 

To get you flowing I am going to create a new branch called `feature/dans-reference`. In here I am going to set up the controller for you but add comments to explain what I have done. Once you are familiar with these changes, merge in the branch `dans-reference` into this one (`feature/dan-api`)